src/inc/perfjitdump.h

```diff
@@ -0,0 +1,29 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+// ===========================================================================
+
+#ifndef PERF_JITDUMP_H
+#define PERF_JITDUMP_H
+
+struct PerfJitDumpState;
```

Comment:
All things implemented in PAL and exposed for the runtime are in pal.h are just plain C functions. We should do the same here. That means expose something like PAL_PerfJitDumpStart, PAL_PerfJitDumpLogMethod and PAL_PerfJitDumpFinish.